src/App.tsx

```diff
@@ -1,24 +1,219 @@
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, {
+  useEffect,
+  useState,
+  useRef,
+} from 'react';
+import { Todo } from './types/Todo';
+import { StatusState } from './types/StatusState';
+import * as todoService from './api/todos';
+import { ErrorNotification } from './components/ErrorNotification';
+import { TodoItem } from './components/TodoItem';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoFooter } from './components/TodoFooter';
+import {
+  UNABLE_ADD_TODO,
+  UNABLE_DELETE_TODO,
+  UNLOADED_TODO,
+} from './utils/constans';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [statusTodo, setStatusTodo] = useState(StatusState.All);
+
+  const [processingTodoIds, setProcessingTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const handleChangeFilter = (newElement: StatusState) => {
+    setStatusTodo(newElement);
+  };
+
+  useEffect(() => {
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(UNLOADED_TODO);
+      });
+  }, []);
+
+  const timerId = useRef<number>(0);
+
+  useEffect(() => {
+    if (timerId.current) {
+      window.clearTimeout(timerId.current);
+    }
+
+    timerId.current = window.setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+  }, [errorMessage]);
+
+  const handleAddTodo = (todoTitle: string) => {
+    setTempTodo({
+      id: 0,
+      title: todoTitle,
+      userId: 0,
+      completed: false,
+    });
+
+    return todoService.addTodo(todoTitle)
+      .then((newTodo) => {
+        setTodos((prevTodos:Todo[]) => [...prevTodos, newTodo]);
+      })
+      .catch(() => {
+        setErrorMessage(UNABLE_ADD_TODO);
+        throw new Error(UNABLE_ADD_TODO);
+      })
+      .finally(() => {
+        setTempTodo(null);
+      });
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    setProcessingTodoIds((prevTodoIds) => [...prevTodoIds, todoId]);
+    todoService.deleteTodo(todoId)
+      .then((() => {
+        setTodos((prevTodos) => prevTodos.filter(todo => todo.id !== todoId));
+      }))
+      .catch(() => {
+        setErrorMessage(UNABLE_DELETE_TODO);
+      })
+      .finally(() => {
+        setProcessingTodoIds(
+          (prevTodoIds) => prevTodoIds.filter(id => id !== todoId),
+        );
+      });
+  };
+
+  const CompletedTodosCount = todos.filter(({ completed }) => completed).length;
+
+  const filteredTodos = todos.filter(todo => {
+    switch (statusTodo) {
+      case StatusState.Active:
+        return !todo.completed;
+      case StatusState.Completed:
+        return todo.completed;
+      case StatusState.All:
+      default:
+        return todo;
+    }
+  });
+
+  const handleUpdateTodo = (todo: Todo, newTodoTitle: string) => {
```

Comment:
`handleUpdateTodo` and `handleToogleTodo` have very similar logic, let's combine them